### ISSUE TITLE
No clear migration path for lift-json library (Scala 2.10 -> 2.12)

### ISSUE THREAD

**Issue by mchwedczuk-box-com** [lift/framework#1995]
Upgrading lift-json library from Scala 2.10 to Scala 2.12 seems more difficult than necessary. The problem is that there is no common lift-json version that is released for both Scala 2.10 and 2.12. 

The highest lift-json library version for Scala 2.10 is `3.0-M1` (https://search.maven.org/artifact/net.liftweb/lift-json_2.10),
the lowest version of the library for Scala 2.12 is `3.0.1` (https://search.maven.org/artifact/net.liftweb/lift-json_2.12).

Not only there is no common version but the above versions seem to be incompatible (e.g. value findField is not a member of net.liftweb.json.JsonAST.JValue), so the semantic version convention is also broken :(

Inspiring article: https://mungingdata.com/scala/maintenance-nightmare-upgrade/

In short you can make life much easier for developers that are "in the tail" and only now perform upgrade to Scala 2.12. If only you release one common version targeting both Scala 2.10 and 2.12. That would allow for smooth migration where you first change only Scala version and make sure that everything works, and then move on to upgrading libraries one by one.

**Comment by farmdawgnation** [comment]
Hi there. I hear you and am sympathetic to the pain point here, but at this point Lift is getting resuscitated after a long hibernation, and we're not likely to go back and make 3.0.1 compatible with Scala 2.10. I remember being a part of the original discussions to drop 2.10 support, but I don't entirely recall _why_ we did that.

In any event, we're looking heavily at decomming lift-json entirely as a part of the recent revival efforts. You may want to look at json4s, which was forked from lift-json back in the day.
